src/main/scala/uncore/tilelink/Crossings.scala

```diff
@@ -0,0 +1,122 @@
+package uncore.tilelink
+
+import Chisel._
+import junctions._
+
+object AsyncClientUncachedTileLinkCrossing {
```

Comment:
It was suggested to me by @terpstra , and I think it's good practice, that the Crossing types should go with the thing that they are crossing, not in this generic Crossing library. For example, the DebugCrossing is in the same file as the DebugModule.  The analogy is that each class can describe a toString() method, you don't put every class' toString method with the String functions.